src/Kmd.Logic.Gateway.Automation/PreValidation/ApiPreValidation.cs

```diff
@@ -87,5 +88,10 @@ public Task<bool> ValidateAsync(GatewayDetails gatewayDetails)
 
             return Task.FromResult<bool>(isValidationSuccess);
         }
+
+        public List<PublishResult> PublishResults
```

Comment:
I would rather return some immutable list to outside. Maybe `IEnumerable`

Reply:
Changed as suggested.